src/test/ui/feature-gates/feature-gate-const_transmute.rs

```diff
@@ -8,4 +8,10 @@ const TRANSMUTED_U32: u32 = unsafe { mem::transmute(Foo(3)) };
 const fn transmute_fn() -> u32 { unsafe { mem::transmute(Foo(3)) } }
 //~^ ERROR can only call `transmute` from const items, not `const fn`
 
+const fn transmute_fn_intrinsic() -> u32 { unsafe { std::intrinsics::transmute(Foo(3)) } }
+//~^ ERROR can only call `transmute` from const items, not `const fn`
```

Comment:
What about outside `unsafe` blocks in a `unsafe const fn` (or maybe entirely outside `unsafe`)?

Reply:
that's unaffected by const fn. It's a completely different source path that checks for that, and it completely ignores the `const` modifier on functions.

Reply:
Sure. But it's cheat to add tests for that here, isn't it?

Reply:
:shrug: I added some